[Dialogs] Fix incorrect vertical layout for uppercase buttons

galiak11 opened the pull request:
# Description

Making sure layout subviews calculates content size based on the available size, not based on the size of the previous layout calculations. This fixes an issue in which the dialog unexpectedly changed vertical vs horizontal actions layout after the first layout pass (see issue #8434 - b/137070719).

## Before
Actions are laid out vertically - when loaded for the first time, and horizontally thereafter (for instance: after orientation changed, app is sent to the background, etc):

![dialog - actions - uppercase - BEFORE](https://user-images.githubusercontent.com/2329102/76383815-6bb92880-6333-11ea-984d-0e392d433ed4.png)

## After
Actions are laid out horizontally on all layout passes - including the first:

![dialog - actions - uppercase - AFTER](https://user-images.githubusercontent.com/2329102/76383817-6c51bf00-6333-11ea-9892-828abc3d7854.png)

## Issues
Closes #8434.
b/137070719, cl/299735842

jverkoey commented on components/Dialogs/src/MDCAlertController.m:
```diff
@@ -563,7 +563,7 @@ - (void)viewWillLayoutSubviews {
       CGRectGetHeight(self.view.bounds) != _previousLayoutSize.height) {
     CGSize currentPreferredContentSize = self.preferredContentSize;
-    CGSize contentSize = CGRectStandardize(self.alertView.bounds).size;
+    CGSize contentSize = CGRectInfinite.size;
```
Does the calculated preferred size eventually get constrained to the viewport's bounds? If so, then this change LGTM, otherwise I'd expect some consideration for the viewport bounds to be added as part of this PR.

galiak11 replied:
This only determines the preferredContentSize, which can be larger than the viewport. The actual bounds are determined by the presentation controller, and that's where the bounds are constrained in case the preferredContentSize is too large.

jverkoey reviewed (commented):
Please add either a unit test or a snapshot test (preferably), or an example that can be used to test this change.

jverkoey reviewed (dismissed):
Marking changes requested for snapshot test.

galiak11 commented:
Snapshot test ready for review - https://github.com/material-components/material-components-ios/pull/9875

Affected snapshot test was added to this PR. Please re-review.

material-automation commented:
bazel detected changes to the following targets:

```
//components/Dialogs:snapshot_tests
//components/Dialogs:snapshot_tests_test_bundle
//components/Dialogs:unit_tests
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Dialogs:unit_tests_environment
//components/Dialogs:unit_tests_environment_test_bundle
```

<!-- identifier: affected-targets -->

galiak11 committed:
[Dialogs] Fix incorrect vertical layout for uppercase buttons

Making sure layout subviews calculates content size based on the available size, not based on the size of the previous layout calculations. This fixes an issue in which the dialog unexpectedly changed vertical vs horizontal actions layout after the first layout pass (see b/137070719).
PiperOrigin-RevId: 299735842
